src/autopas/LogicHandler.h

```diff
@@ -377,26 +384,27 @@ class LogicHandler {
   }
 
   /**
-   * This function covers the full pipeline of all mechanics happening during the pairwise iteration.
+   * This function covers the full pipeline of all mechanics happening during the computation of particle interactions.
    * This includes:
    * - selecting a configuration
    *   - gather live info, homogeneity, and max density
    *   - get next config (tuning)
    *   - check applicability
    *   - instantiation of traversal and container
    * - triggering iteration and tuning result logger
-   * - pairwise iteration
+   * - computing the interactions
    *   - init and end traversal
    *   - remainder traversal
    *   - measurements
    * - pass measurements to tuner
    *
    * @tparam Functor
    * @param functor
+   * @param interactionType
    * @return True if this was a tuning iteration.
    */
   template <class Functor>
-  bool iteratePairwisePipeline(Functor *functor);
+  bool computeInteractionsPipeline(Functor *functor, InteractionTypeOption::Value interactionType);
```

Comment:
```suggestion
  bool computeInteractionsPipeline(Functor *functor, const InteractionTypeOption &interactionType);
```
Why only take the value? If you have the full object, you get the utility functions and e.g. things like error messages are easier to pretty-print.
I won't mark all occurrences so please check yourself where this pattern is used.

Reply:
Just my lack of understanding of these option enums.
I think I understand them better now :D Changed it and thanks